SwiftGit2Tests/RepositorySpec.swift

```diff
@@ -429,5 +429,71 @@ class RepositorySpec: QuickSpec {
 				expect(result.value as? Reference).notTo(beNil())
 			}
 		}
+		
+		describe("Repository.setHEAD(OID)") {
+			it("should set HEAD to the OID") {
+				let repo = Fixtures.simpleRepository
+				let oid = OID(string: "315b3f344221db91ddc54b269f3c9af422da0f2e")!
+				expect(repo.HEAD().value?.shortName).to(equal("master"))
+				
+				expect(repo.setHEAD(oid)).to(haveSucceeded())
+				let HEAD = repo.HEAD().value
+				expect(HEAD?.longName).to(equal("HEAD"))
+				expect(HEAD?.oid).to(equal(oid))
+				
+				expect(repo.setHEAD(repo.localBranchWithName("master").value!)).to(haveSucceeded())
+				expect(repo.HEAD().value?.shortName).to(equal("master"))
+			}
+		}
+		
+		describe("Repository.setHEAD(ReferenceType)") {
+			it("should set HEAD to a branch") {
+				let repo = Fixtures.detachedHeadRepository
+				let oid = repo.HEAD().value!.oid
+				expect(repo.HEAD().value?.longName).to(equal("HEAD"))
+				
+				let branch = repo.localBranchWithName("another-branch").value!
+				expect(repo.setHEAD(branch)).to(haveSucceeded())
+				expect(repo.HEAD().value?.shortName).to(equal(branch.name))
+				
+				expect(repo.setHEAD(oid)).to(haveSucceeded())
+				expect(repo.HEAD().value?.longName).to(equal("HEAD"))
+			}
+		}
+		
+		describe("Repository.checkout()") {
+			// We're not really equipped to test this yet. :(
```

Comment:
nitpick: Perhaps use `xdescribe` with `it` closures explaining what you would test, if you could?